application/backend/src/main/resources/apps/typerefinery/components/layout/treeview/clientlibs/functions.js

```diff
@@ -107,7 +109,47 @@ Typerefinery.Components.Widgets.Treeview.Instance = Typerefinery.Components.Widg
         });
     };
 
+    ns.onFilterNode = ($component, data) => {
+        console.log("onFilterNode", data)
+        const componentConfig = componentNs.getComponentConfig($component);
+        const treeViewInstance = treeViewInstanceNs[componentConfig.id];
+        treeViewInstance.treeview('collapseAll', { silent: true });
+        treeViewInstance.treeview('enableAll', { silent: true });
+        treeViewInstance.treeview('search', [data.value, {
+            ignoreCase: true,     // case insensitive
+            exactMatch: false,    // like or equals
+            revealResults: true,  // reveal matching nodes
+        }]);
+    };
+
+
+    ns.highlightComponent =  ($component, data) => {
+        console.log("highlightComponent", data)
+        const componentConfig = componentNs.getComponentConfig($component);
+        const treeViewInstance = treeViewInstanceNs[componentConfig.id];
+        treeViewInstance.treeview('collapseAll', { silent: true });
+        treeViewInstance.treeview('enableAll', { silent: true });
+        treeViewInstance.treeview('search', [data.value, {
+            ignoreCase: true,     // case insensitive
+            exactMatch: false,    // like or equals
+            revealResults: true,  // reveal matching nodes
+        }]);
+    };
+
+    ns.getEventHandlerCallBackFn = ($component, event) => {
+        if(event.topic && event.type === 'FILTER') {
```

Comment:
Can we turn this into config component level map you can pick events from it?

{ 
"FILTER": ns.eventOnFilter,
"HIGHLIGHT": ns.eventOnHighlight
}


This way when next event is added its just a new function plus config update?